odxtools/minmaxlengthtype.py

```diff
@@ -34,113 +34,136 @@ def __post_init__(self):
+                              f"must not be longer than {self.max_length} bytes. "
+                              f"(Is: {len(value_bytes)} bytes.)")
+
+        return value_bytes
 
     def convert_bytes_to_internal(self, decode_state: DecodeState, bit_position: int = 0):
         if decode_state.cursor_position + self.min_length > len(decode_state.coded_message):
-            raise DecodeError("The PDU ended before min length was reached.")
+            raise DecodeError("The PDU ended before minimum length was reached.")
 
         coded_message = decode_state.coded_message
-        cursor_position = decode_state.cursor_position
-        termination_char = self.__termination_character()
+        cursor_pos = decode_state.cursor_position
+        termination_seq = self.__termination_sequence()
 
-        # If no termination char is found, this is the next byte after the parameter.
-        max_termination_byte = len(coded_message)
+        max_terminator_pos = len(coded_message)
         if self.max_length is not None:
-            max_termination_byte = min(max_termination_byte, cursor_position + self.max_length)
+            max_terminator_pos = min(max_terminator_pos, cursor_pos + self.max_length)
 
         if self.termination != "END-OF-PDU":
-            # The parameter either ends after max length, at the end of the PDU
-            # or if a termination character is found.
-            char_length = len(termination_char)  # either 1 or 2
-
-            termination_byte = cursor_position + self.min_length
-            found_char = False
-            # Search the termination character
-            while termination_byte < max_termination_byte and not found_char:
-                found_char = (
-                    coded_message[termination_byte:termination_byte +
-                                  char_length] == termination_char)
-                if not found_char:
-                    termination_byte += char_length
-
-            byte_length = termination_byte - cursor_position
+            # The parameter either ends after the maximum length, at
+            # the end of the PDU or if a termination sequence is
+            # found.
+
+            terminator_pos = cursor_pos + self.min_length
+            while True:
+                # Search the termination sequence
+                terminator_pos = coded_message.find(termination_seq, terminator_pos,
+                                                    max_terminator_pos)
+                if terminator_pos < 0:
+                    # termination sequence was not found, i.e., we
+                    # are terminated by either the end of the PDU or
+                    # our maximum size. (whatever is the smaller
+                    # value.)
+                    terminator_pos = len(coded_message)
+                    if self.max_length is not None:
+                        terminator_pos = min(cursor_pos + self.max_length, terminator_pos)
```

Comment:
Why not just `terminator_pos = max_terminator_pos` ?

Reply:
good point.  fixed: f7edde6386ce41ef08d5a16caecad4a224c58860